torch/_inductor/template_heuristics/aten.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import annotations
+
+from typing import Any, TYPE_CHECKING
+
+from torch._inductor import config as inductor_config
+
+from ..kernel.bmm import aten_baddbmm, aten_bmm, aten_bmm_dtype
+from ..kernel.mm import aten__fp8_mm, aten__int_mm, aten_addmm, aten_bias_addmm, aten_mm
+from ..kernel.mm_plus_mm import aten_mm_plus_mm
+from .base import TemplateConfigHeuristics
+
+
+if TYPE_CHECKING:
+    from collections.abc import Generator
+
+    from ..ir import Layout
+    from ..kernel_inputs import KernelInputs
+
+from .registry import register_template_heuristic
+
+
+@register_template_heuristic(aten_mm.uid, "cuda")
+@register_template_heuristic(aten_mm.uid, "cpu")
+@register_template_heuristic(aten_mm.uid, "xpu")
+@register_template_heuristic(aten_mm.uid, "mtia")
```

Comment:
Maybe ass support for something like:
```suggestion
@register_template_heuristic(aten_mm.uid, ["cuda", "cpu", "xpu", "mtia"])
```
to avoid duplication

Reply:
Makes sense! I actually refactored this in another PR, so that we support saying "None" for device type filtering to indicate that it's valid for all device types. So ATEN looks cleaner now, with just `e.g. @register_template_heuristic(aten_mm.uid, None)`